app/src/main/java/com/prudhvir3ddy/todo_app_gettingthingsdone/view/splash/SplashActivity.kt

```diff
@@ -54,4 +69,85 @@ class SplashActivity : AppCompatActivity() {
       })
 
   }
-}
+
+  override fun onResume() {
+    super.onResume()
+    //check for update after retrieving token
+    Handler().postDelayed(this::triggerUpdate, 1000)
+  }
+
+  private fun triggerUpdate() {
+    appUpdateInfoTask.addOnSuccessListener { appUpdateInfo ->
+      if (appUpdateInfo.installStatus() == InstallStatus.DOWNLOADED) {
+        popSnackbarForUpdate()
+      } else if (appUpdateInfo.updateAvailability() == UpdateAvailability.DEVELOPER_TRIGGERED_UPDATE_IN_PROGRESS) {
+        try {
+          appUpdateManager.startUpdateFlowForResult(
+            appUpdateInfo,
+            IMMEDIATE,
+            this,
+            1011
+          )
+        } catch (e: SendIntentException) {
+          e.printStackTrace()
+        }
+      } else if (appUpdateInfo.updateAvailability() == UpdateAvailability.UPDATE_AVAILABLE && appUpdateInfo.isUpdateTypeAllowed(
+          IMMEDIATE
+        )
+      ) {
+        appUpdateManager.startUpdateFlowForResult(
+          appUpdateInfo,
+          IMMEDIATE,
+          this,
+          1001
+        )
+      }
+    }
+  }
+
+  private val updatedListener: InstallStateUpdatedListener = object : InstallStateUpdatedListener {
```

Comment:
are you supposed to use this somewhere ?

updatedListener

Reply:
This method, has never been called explicitly but indeed overrides a method which obviously is executed. So instead of making the method anonymous, I preferred giving it a name(only for sake of simplicity and better legibility)